module/VuFind/src/VuFind/Db/Row/ResourceTags.php

```diff
@@ -49,4 +75,137 @@ public function __construct($adapter)
     {
         parent::__construct('id', 'resource_tags', $adapter);
     }
+
+    /**
+     * Get identifier (returns null for an uninitialized or non-persisted object).
+     *
+     * @return ?int
+     */
+    public function getId(): ?int
+    {
+        return $this->id ?? null;
+    }
+
+    /**
+     * Get resource.
+     *
+     * @return ResourceEntityInterface
+     */
+    public function getResource(): ResourceEntityInterface
+    {
+        return $this->resource_id
+        ? $this->getDbServiceManager()->get(ResourceServiceInterface::class)->getResourceById($this->resource_id)
+        : null;
+    }
+
+    /**
+     * Set resource.
+     *
+     * @param ResourceEntityInterface $resource Resource
+     *
+     * @return ResourceTagsEntityInterface
+     */
+    public function setResource(ResourceEntityInterface $resource): ResourceTagsEntityInterface
+    {
+        $this->resource_id = $resource->getId();
```

Comment:
This should be `?->getId()` once the parameter is made nullable (as should the other impacted setters).